src/NzbDrone.Core/MediaFiles/EpisodeFileMovingService.cs

```diff
@@ -113,7 +117,33 @@ private EpisodeFile TransferFile(EpisodeFile episodeFile, Series series, List<Ep
                 throw new SameFilenameException("File not moved, source and destination are the same", episodeFilePath);
             }
 
-            _diskTransferService.TransferFile(episodeFilePath, destinationFilePath, mode);
+            bool needsRename = false;
+            bool transfer = true;
+
+            if (scriptImportDecisionInfo is not null)
```

Comment:
I think this logic might be better suited for `UpgradeMediaFileService`, it'll avoid having to pass information back to it and it already knows whether it's going to be a move or copy.